modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/util/CPInstanceHelperImpl.java

```diff
@@ -394,6 +394,16 @@ public FileVersion getCPInstanceImageFileVersion(
 			return null;
 		}
 
+		CPDefinition cpDefinition = _cpDefinitionLocalService.getCPDefinition(
+			cpInstance.getCPDefinitionId());
+
+		if (_commerceProductViewPermission.contains(
```

Comment:
Is this check right? SHouldn't be ! contains?

Reply:
You're right...fixed!